doc/devel/development.md

```diff
@@ -0,0 +1,1932 @@
+middlewares are applied before the information is sent to the Context Broker, modifiying the entity before it is sent to
+Orion. The queryContext middlewares are applied on the received data, whenever the IoT Agent queries the Context Broker
+for information. I.e.: both middlewares will be automatically applied whenever the `update()` or `query()` functions are
+called in the library.
+
+All the middlewares have the opportunity to break the chain of middleware applications by calling the `callback()` with
+an error object (the usual convention). If any of the updateContext middlewares raise an error, no request will be sent
+to the Context Broker. On the other hand, the queryContext request is always performed, but the call to the `query()`
+function will end up in an error if any of the queryContext middlewares report an error.
+
+### Plugins usage
+
+All the middlewares have the same signature:
+
+```javascript
+function middlewareName(entity, typeInformation, callback) {}
+```
+
+The arguments for any middleware are the NGSI data over which it can operate:
+
+-   An updateContext payload in the case of an updateContext middleware and a queryContext payload otherwise;
+-   a typeInformation object containing all the information about the device stored during registration.
+-   and the customary `callback` parameter, with the usual meaning. It's really important for the library user to call
+    this callback, as failing to do so may hang the IoT Agent completely. The callback must be called with the an
+    optional error in the first argument and the same arguments received (potentially modified) as the following.
+
+In order to manage the middlewares to the system, the following functions can be used:
+
+-   `addUpdateMiddleware`: adds an updateContext middleware to the stack of middlewares. All the middlewares will be
+    applied to every call to the `update()` function. The final payload of the updateContext request will be the result
+    of applying all this middlewares in the order they have been defined.
+
+-   `addQueryMiddleware`: adds a queryContext middleware to the stack of middlewares. All the middlewares will be
+    applied to every call to the `query()` function.
+
+-   `resetMiddlewares`: remove all the middlewares from the system.
+
+Usually, the full list of middlewares an IoT Agent will use would be added in the IoTAgent start sequence, so they
+should not change a lot during the IoT lifetime.
+
+### Provided plugins
+
+The library provides some plugins out of the box, in the `dataPlugins` collection. In order to load any of them, just
+use the `addQueryMiddleware` and `addUpdateMiddleware` functions with the selected plugin, as in the example:
+
+```javascript
+var iotaLib = require('iotagent-node-lib');
+
+iotaLib.addUpdateMiddleware(iotaLib.dataPlugins.compressTimestamp.update);
+iotaLib.addQueryMiddleware(iotaLib.dataPlugins.compressTimestamp.query);
+```
+
+## Developing a new IoT Agent
```

Comment:
Has this section walked through in order to check if the existing instructions are still valid to create a new IOTA?

If that has not been done and it is out of scope of this PR, at least a WARNING should be included about the potential obsolescence of the procedure.

Reply:
Your're right, not the scope of this PR. Added a warning note in https://github.com/telefonicaid/iotagent-node-lib/pull/1371/commits/e46c18c1072247932b22ace41306a0c301588c23

Reply:
This is going to be done in a subsequent PR. Noted here: https://github.com/telefonicaid/iotagent-node-lib/issues/1329#issue-1643605738
NTC